leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/QueueModeTest.java

```diff
@@ -0,0 +1,200 @@
+        queueModeHelper.waitForUpdate(SLEEPTIME * 1010);
+
+        // Check that client is awake
+        queueModeHelper.ensureClientAwake();
+        queueModeHelper.resetAwakeLatch();
+
+        // Wait for client awake time expiration (1% margin)
+        queueModeHelper.waitForAwakeTime(AWAKETIME * 1010);
+
+        // Check that client is sleeping
+        queueModeHelper.ensureClientSleeping();
+
+        // Stop client with out de-registration
+        queueModeHelper.client.stop(false);
+    }
+
+    @Test
+    public void one_awake_notification() {
+
+        // Check client is not registered
+        queueModeHelper.assertClientNotRegisterered();
+
+        // Start it and wait for registration
+        queueModeHelper.client.start();
+        queueModeHelper.waitForRegistration(1);
+
+        // Check client is well registered
+        queueModeHelper.assertClientRegisterered();
+
+        // Check that client is awake and only one awake notification
+        queueModeHelper.ensureClientAwake();
+        queueModeHelper.ensureOneAwakeNotification();
+
+        // Triggers one update
+        queueModeHelper.client.triggerRegistrationUpdate();
+        queueModeHelper.waitForUpdate(1000);
+        queueModeHelper.resetLatch();
+
+        // Check only one notification
+        queueModeHelper.ensureOneAwakeNotification();
+
+        // Wait for client awake time expiration (1% margin)
+        queueModeHelper.waitForAwakeTime(AWAKETIME * 1010);
+
+        // Check that client is sleeping
+        queueModeHelper.ensureClientSleeping();
+
+        // Wait for update when waking up (1% margin)
+        queueModeHelper.waitForUpdate(SLEEPTIME * 1010);
+
+        // Check that client is awake
+        queueModeHelper.ensureClientAwake();
+        queueModeHelper.resetAwakeLatch();
+
+        // Triggers two updates
+        queueModeHelper.client.triggerRegistrationUpdate();
+        queueModeHelper.waitForUpdate(1010);
+        queueModeHelper.resetLatch();
+        queueModeHelper.client.triggerRegistrationUpdate();
+        queueModeHelper.waitForUpdate(1010);
+        queueModeHelper.resetLatch();
+
+        // Check only one notification
+        queueModeHelper.ensureOneAwakeNotification();
+
+    }
+
+    @Test
+    public void sleeping_if_timeout() throws InterruptedException {
+
+        // Check client is not registered
+        queueModeHelper.assertClientNotRegisterered();
+
+        // Start it and wait for registration
+        queueModeHelper.client.start();
+        queueModeHelper.waitForRegistration(1);
+
+        // Check client is well registered and awake
+        queueModeHelper.assertClientRegisterered();
+        queueModeHelper.ensureClientAwake();
+
+        // Send a response with very short timeout so that exception is thrown
+        ReadResponse response = queueModeHelper.server.send(queueModeHelper.getCurrentRegistration(),
+                new ReadRequest(3, 0, 1), 1);
```

Comment:
Should we increase it then? What value is more suitable in your computer?

Reply:
If I well understand you choose a very small timeout. In fact the smaller one : 1 ms.
Hoping that the request will timeout before we get the response.

My computer seems too fast :p because this tests fail almost all the time, so I think we should find another strategy.

Maybe stoping the client without deregistration ?